pyaerocom/aeroval/setupclasses.py

```diff
@@ -223,112 +220,215 @@ def _get_all_period_strings(self):
+    @computed_field
+    @property
+    def colocation_opts(self) -> ColocationSetup:
+        if hasattr(self, "model_extra") & bool(
+            cfg_extra_keys := set(self.model_extra).intersection(
+                set(ColocationSetup().__dict__.keys())
+            )
+        ):
+            subset_dict = {k: self.model_extra[k] for k in cfg_extra_keys}
+            # need to pass some default values to the ColocationSetup if not provided in config
+            default_dict = {"save_coldata": True, "keep_data": False, "resample_how": "mean"}
+            for key in default_dict:
+                if key not in subset_dict:
+                    subset_dict[key] = default_dict[key]
+
+            return ColocationSetup(**subset_dict)
+        else:
+            return ColocationSetup(save_coldata=True, keep_data=False, resample_how="mean")
+
+    @field_serializer("colocation_opts")
+    def serialize_colocation_opts(self, colocation_opts: ColocationSetup):
+        return colocation_opts.json_repr()
+
+    @computed_field
+    @property
+    def webdisp_opts(self) -> WebDisplaySetup:
+        if hasattr(self, "model_extra") & bool(
+            cfg_extra_keys := set(self.model_extra).intersection(set(WebDisplaySetup.model_fields))
+        ):
+            subset_dict = {k: self.model_extra[k] for k in cfg_extra_keys}
+            return WebDisplaySetup(**subset_dict)
+        else:
+            return WebDisplaySetup()
+
+    @computed_field
+    @property
+    def processing_opts(self) -> EvalRunOptions:
+        if hasattr(self, "model_extra") & bool(
+            cfg_extra_keys := set(self.model_extra).intersection(set(EvalRunOptions.model_fields))
+        ):
+            subset_dict = {k: self.model_extra[k] for k in cfg_extra_keys}
+            return EvalRunOptions(**subset_dict)
+        else:
+            return EvalRunOptions()
+
+    var_web_info: dict = {}
+
+    @computed_field
+    @property
+    def statistics_opts(self) -> StatisticsSetup:
+        if hasattr(self, "model_extra") & bool(
+            cfg_extra_keys := set(self.model_extra).intersection(set(StatisticsSetup.model_fields))
+        ):
+            subset_dict = {k: self.model_extra[k] for k in cfg_extra_keys}
+            return StatisticsSetup(**subset_dict)
+        else:
+            return StatisticsSetup(weighted_stats=True, annual_stats_constrained=False)
+
+    # ObsCollection and ModelCollection require special attention b/c they're not based on Pydantic.
+    # TODO Use Pydantic for ObsCollection and ModelCollection
+    obs_cfg: ObsCollection | dict = ObsCollection()
+
+    @field_validator("obs_cfg")
+    def validate_obs_cfg(cls, v):
+        if not isinstance(v, ObsCollection):
+            return ObsCollection(v)
+        return v
+
+    @field_serializer("obs_cfg")
+    def serialize_obs_cfg(self, obs_cfg: ObsCollection):
+        return obs_cfg.json_repr()
+
+    model_cfg: ModelCollection | dict = ModelCollection()
```

Comment:
`model_cfg` and `obs_cfg` are lumped together at the bottom because they are based on `ObsCollection` and `ModelCollection` which are not (yet) based on pydantic. I have added comments to more clearly outline which parts of the code are pydantic-based attributes, non-pydantic-base attributes, and methods.